pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -0,0 +1,300 @@
+
+const (
+	ModePatch Mode = "patch"
+	ModeSSA   Mode = "ssa"
+)
+
+// Action deploys the resources that are included in the ReconciliationRequest using
+// the same create or patch machinery implemented as part of deploy.DeployManifestsFromPath.
+type Action struct {
+	fieldOwner  string
+	deployMode  Mode
+	labels      map[string]string
+	annotations map[string]string
+}
+
+type ActionOpts func(*Action)
+
+func WithFieldOwner(value string) ActionOpts {
+	return func(action *Action) {
+		action.fieldOwner = value
+	}
+}
+func WithMode(value Mode) ActionOpts {
+	return func(action *Action) {
+		action.deployMode = value
+	}
+}
+
+func WithLabel(name string, value string) ActionOpts {
+	return func(action *Action) {
+		if action.labels == nil {
+			action.labels = map[string]string{}
+		}
+
+		action.labels[name] = value
+	}
+}
+
+func WithLabels(values map[string]string) ActionOpts {
+	return func(action *Action) {
+		if action.labels == nil {
+			action.labels = map[string]string{}
+		}
+
+		for k, v := range values {
+			action.labels[k] = v
+		}
+	}
+}
+
+func WithAnnotation(name string, value string) ActionOpts {
+	return func(action *Action) {
+		if action.annotations == nil {
+			action.annotations = map[string]string{}
+		}
+
+		action.annotations[name] = value
+	}
+}
+
+func WithAnnotations(values map[string]string) ActionOpts {
+	return func(action *Action) {
+		if action.annotations == nil {
+			action.annotations = map[string]string{}
+		}
+
+		for k, v := range values {
+			action.annotations[k] = v
+		}
+	}
+}
+
+func (a *Action) run(ctx context.Context, rr *odhTypes.ReconciliationRequest) error {
+	for i := range rr.Resources {
+		obj := rr.Resources[i]
+
+		current, lookupErr := a.lookup(ctx, rr.Client, obj)
+		if lookupErr != nil && !k8serr.IsNotFound(lookupErr) {
+			return fmt.Errorf("failed to lookup object %s/%s: %w", obj.GetNamespace(), obj.GetName(), lookupErr)
+		}
+
+		resources.SetLabels(&obj, a.labels)
+		resources.SetAnnotations(&obj, a.annotations)
+
+		resources.SetAnnotation(&obj, annotations.ComponentGeneration, strconv.FormatInt(rr.Instance.GetGeneration(), 10))
+		resources.SetAnnotation(&obj, annotations.PlatformType, string(rr.Release.Name))
+		resources.SetAnnotation(&obj, annotations.PlatformVersion, rr.Release.Version.String())
+
+		switch {
+		// the user has explicitly marked the current object as not owned by the operator, so
+		// skip any further processing
+		case current != nil && resources.GetAnnotation(current, annotations.ManagedByODHOperator) == "false":
+			continue
+
+		// The object is explicitly marked as not owned by the operator in the manifests,
+		// so it should be created if it doesn't exist, but should not be modified afterward.
+		case resources.GetAnnotation(&obj, annotations.ManagedByODHOperator) == "false":
+			// remove the opendatahub.io/managed as it should not be set
+			// to the actual object in this case
+			resources.RemoveAnnotation(&obj, annotations.ManagedByODHOperator)
```

Comment:
not sure i understand this:
if the resource exist in the cluster, and with annoation to false, we dont touch it
if the resource does not exist, (like we just enable component) and component team mark the annoation in manifests with annotation to false, we remove that annoation then create it?  then next time run into "run" it will for intl the first swtich...case ?

Reply:
it is the same for each reconciliation. the annotation is removed as the behavior is controller by the component developers, not the user so I didn't want it to be set on the cluster. But is something we can talk about since that annotation is for the moment only an internal annotation (i.e. it is added to the `OdhDashboardConfig`)